xarray/core/dataarray.py

```diff
@@ -811,12 +812,8 @@ def rename(self, new_name_or_name_dict):
         """
         if utils.is_dict_like(new_name_or_name_dict):
             name_dict = new_name_or_name_dict.copy()
```

Comment:
I have pushed another commit fa6b02c to remove it. Keeping the PR open for another while is fine by me.